contracts/modules/DToken.sol

```diff
@@ -39,9 +39,10 @@ contract DToken is BaseLogic {
         return string(abi.encodePacked("d", IERC20(underlying).symbol()));
     }
 
-    /// @notice Decimals, always normalised to 18.
-    function decimals() external pure returns (uint8) {
-        return 18;
+    /// @notice Decimals of underlying
+    function decimals() external view returns (uint8) {
+        (address underlying,,,) = CALLER();
+        return IERC20(underlying).decimals();
```

Comment:
Any advantage to getting this from AssetCache? Probably costs more gas slightly, but at least it won't change... Probably this is the best way, but let's discuss it just in case.

Reply:
fixed